File: src/main/java/de/istkorrekt/gameoflife/MatchfieldFactory.java
package de.istkorrekt.gameoflife;

import java.util.Collection;
import java.util.Collections;

public class MatchfieldFactory {

    public Matchfield createAllDead(Size size) {
        return createWithAlive(Collections.emptyList(), size);
    }

    public Matchfield createWithAlive(Collection<Location> locations, Size size) {
        return new ArrayMatchfield(locations, size);
    }
}

File: src/main/kotlin/de/istkorrekt/gameoflife/MatchfieldFactory.kt
package de.istkorrekt.gameoflife

object MatchfieldFactory {

    fun createAllDead(size: Size): Matchfield {
        return ArrayMatchfield(size)
    }

    fun createWithAlive(size: Size, aliveLocations: Collection<Location>): Matchfield {
        return ArrayMatchfield(size, aliveLocations)
    }
}